File: 백준/Gold/2470. 두 용액/두 용액.java
import java.io.*;
import java.util.*;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		
		int n = Integer.parseInt(bf.readLine());
		int[] arr = new int[n];
		
		st = new StringTokenizer(bf.readLine());
		for(int i=0; i<n; i++) {
			arr[i] = Integer.parseInt(st.nextToken());
		}
		
		
		Arrays.sort(arr);
//		System.out.println(Arrays.toString(arr));
		
		
		int a = 0;
		int b = 0;
		
		int left = 0;
		int right = n-1;
		int min = Integer.MAX_VALUE;
		
		while(left < right) {
			int temp = arr[left] + arr[right];
			int sum = Math.abs(arr[left] + arr[right]);
			
			if(min > sum) {
				min = sum;
				a = arr[left];
				b = arr[right];
			}
			
			if(temp < 0) {
				left++;				
			}else {
				right--;
			}
		}
		
		System.out.println(a + " " + b);
	}
}

File: 백준/Gold/2470. 두 용액/두 용액.js
const path = process.platform === 'linux' ? '/dev/stdin' : 'index.txt';
const input = require('fs').readFileSync(path).toString().trim().split('\n');

const N = +input[0];
const arr = input[1].split(' ').map(Number);

arr.sort((a, b) => a - b);

let [a, b] = [0, 0];
const binarySearch = (n, arr) => {
  let left = 0;
  let right = n - 1;
  let min = Infinity;

  while (left < right) {
    const sum = arr[left] + arr[right];
    const absSum = Math.abs(arr[left] + arr[right]);

    if (min > absSum) {
      min = absSum;
      a = arr[left];
      b = arr[right];
    }

    if (sum < 0) {
      left++;
    } else {
      right--;
    }
  }
};

binarySearch(N, arr);
console.log(a, b);